File: abc/005/3.rb
t = gets.to_i
gets
as = gets.split.map(&:to_i)
gets
bs = gets.split.map(&:to_i)

bs.each do |b|
  idx = as.find_index { |a| a <= b and b - a <= t }
  unless idx
    puts "no"
    exit
  end
  as.delete_at(idx)
end
puts "yes"

File: abc/005/3.py
import bisect
T = int(raw_input())
N = int(raw_input())
A = map(int, raw_input().split())
M = int(raw_input())
B = map(int, raw_input().split())

A.sort()
B.sort()
for b in B:
    pos = bisect.bisect_left(A, b-T)
    if (pos == len(A) or A[pos] > b):
        print 'no'
        exit()
    A = A[:pos] + A[pos+1:]
print 'yes'
